File: Services/analyse/src/main/java/org/example/analyse/Models/entities/Biology.java
package org.example.analyse.Models.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;
import org.example.analyse.Models.enums.BiologyInterpretation;
import org.example.analyse.Models.enums.BiologyType;

import java.time.LocalDate;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class Biology {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    long id;
    LocalDate date;
    BiologyType type;
    double value;
    BiologyInterpretation interpretation;
    String comment;
    long consultationId;
}

File: front/src/app/Models/Biology.ts
import { BiologyType, BiologyInterpretation } from "./BiologyEnums";

export class Biology {
  id!: number;
  date!: Date;
  type!: typeof BiologyType[keyof typeof BiologyType];
  value!: number;
  interpretation!: typeof BiologyInterpretation[keyof typeof BiologyInterpretation];
  comment!: string;
  consultationId!: number;
}